bip-0340/reference.py

```diff
@@ -81,38 +90,43 @@ def is_square(x):
     return pow(x, (p - 1) // 2, p) == 1
 
 def has_square_y(P):
-    return not is_infinity(P) and is_square(y(P))
+    return (not is_infinity(P)) and (is_square(y(P)))
 
 def has_even_y(P):
     return y(P) % 2 == 0
 
 def pubkey_gen(seckey):
-    x = int_from_bytes(seckey)
-    if not (1 <= x <= n - 1):
+    d0 = int_from_bytes(seckey)
+    if not (1 <= d0 <= n - 1):
+        debug_print_vars()
         raise ValueError('The secret key must be an integer in the range 1..n-1.')
-    P = point_mul(G, x)
+    P = point_mul(G, d0)
     return bytes_from_point(P)
 
-def schnorr_sign(msg, seckey0, aux_rand):
+def schnorr_sign(msg, seckey, aux_rand):
     if len(msg) != 32:
+        debug_print_vars()
```

Comment:
I think this `debug_print_vars()` is not very helpful and can be removed to reduce lines of code. The main reason for having these is to allow comparing your implementation with the test vectors (presumably you only need one test vector). But none of the test vectors hit this case. If your implementation accepts `len(msg) != 32`, and you give the same inputs to the reference code, you will immediately see where the problem lies thanks to the exception alone. I think this holds for any `debug_print_vars` before raising exceptions.